rpctest/utils.go

```diff
@@ -76,24 +77,25 @@ func syncMempools(nodes []*Harness) error {
 	return nil
 }
 
-// syncBlocks blocks until all nodes report the same block height.
+// syncBlocks blocks until all nodes report the same best chain.
 func syncBlocks(nodes []*Harness) error {
 	blocksMatch := false
 
+retry:
 	for !blocksMatch {
-	retry:
-		blockHeights := make(map[int64]struct{})
+		blocksByHash := make(map[chainhash.Hash]int32)
 
 		for _, node := range nodes {
-			blockHeight, err := node.Node.GetBlockCount()
+			blockHash, blockHeight, err := node.Node.GetBestBlock()
 			if err != nil {
 				return err
 			}
 
-			blockHeights[blockHeight] = struct{}{}
-			if len(blockHeights) > 1 {
+			existingHeight, ok := blocksByHash[*blockHash]
```

Comment:
It's actually used below in the `if len(blocksByHash) > 1 || ok &&` statement.

However, I think you're right that it doesn't really need to be checked since the only time the actual `blockHeight` will be 0 is when it's the genesis block, and if you're on two different networks, the hashes won't be the same so it'll fail the `if len(blocksByHash) > 1` check.